anchore_engine/services/catalog/archiver.py

```diff
@@ -1168,20 +1168,30 @@ def archive_vuln_history(self, img_archive: ImageArchive) -> list:
             logger.exception("Error flushing policy engine state for image")
             raise ex
 
-    # Removed since the archive task should not affect any sources
-    # def flush_source_objects(self, src_mgr, artifacts):
-    #     for artifact in artifacts:
-    #         logger.debug('Flushing source artifact: {}'.format(artifact.name))
-    #         if artifact.source and artifact.source.bucket:
-    #             src_mgr.delete(self.account, artifact.source.bucket, artifact.source.key)
-    #
-    #     with session_scope() as session:
-    #         db_catalog_image.delete(self.image_digest, self.account, session)
-    #
-    # def flush_policy_engine(self, image_id):
-    #     try:
-    #         pe_client = internal_client_for(PolicyEngineClient, userId=self.account)
-    #         pe_client.delete_image(self.account, image_id)
-    #     except Exception as ex:
-    #         logger.exception("Error flushing policy engine state for image")
-    #         raise ex
+
+class MaxImageAnalysesArchiver(object):
```

Comment:
This could just run in a monitor thread in the Catalog service or something if we decide to go async. 

Otherwise, if synchronous we can just fire this off when an image is added